src/runtime/virtcontainers/fs_share.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2022 Apple Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package virtcontainers
+
+import (
+	"context"
+
+	"github.com/kata-containers/kata-containers/src/runtime/virtcontainers/pkg/agent/protocols/grpc"
+)
+
+// fsShareTracingTags defines tags for the trace span
+var fsShareTracingTags = map[string]string{
+	"source":    "runtime",
+	"package":   "virtcontainers",
+	"subsystem": "fs_share",
+}
+
+// SharedFile represents the outcome of a host filesystem sharing
+// operation.
+type SharedFile struct {
+	storage   *grpc.Storage
+	guestPath string
+}
+
+type FilesystemSharer interface {
+	// Prepare will set the host filesystem up, making it ready
+	// to share container files and directories with the guest.
+	// Implementation of this method must be idempotent and be
+	// ready to potentially be called several times in a row,
+	// without symetric calls to Cleanup in between.
+	Prepare(context.Context) error
```

Comment:
Could you give examples of what this could do. I _guess_ it would include unmounting host filesystems and things like that, but I'm not entirely sure that's the intent.

Also, it might be interesting to watch https://github.com/kubernetes/enhancements/pull/2893 while you are working in that space.

Reply:
> Could you give examples of what this could do. I _guess_ it would include unmounting host filesystems and things like that, but I'm not entirely sure that's the intent.
> 

A further commit implements that interface for Linux, so I guess it's a good example :)
I added more comments to briefly describer the Linux case.